### PR TITLE
[Issue 2247] fix datatype of vhost:headers from String to Array

### PR DESCRIPTION
**Pull request by chillinger**
Fix wrong introduction of Dataype String to headers parameter
The code requires the parameter to be Array[String] to work properly

### CODE REVIEW

**Comment by david22swan** [comment]
From what I can see this looks to be correct, but could you also update the example in the description so that it matches

**Comment by chillinger** [comment]
@david22swan thanks for the review, description is updated

**Comment by david22swan** [comment]
@chillinger Look's like your getting some unit test failures.

File: manifests/vhost.pp [hunk 1]
```diff
@@ -1853,7 +1853,7 @@
   Optional[Variant[Array[String],String]] $redirectmatch_status                       = undef,
   Optional[Variant[Array[String],String]] $redirectmatch_regexp                       = undef,
   Optional[Variant[Array[String],String]] $redirectmatch_dest                         = undef,
-  Optional[String] $headers                                                           = undef,
+  Optional[Array[String]] $headers                                                    = undef,
```
Comment (ekohl) [inline manifests/vhost.pp:1856]: If you're changing it to an array, I'd say drop the `Optional` part and make it an empty array by default. I'd also narrow it to non-empty strings:
```suggestion
  Array[String[1]] $headers                                                    = [],
```

Then you can simplify `templates/vhost/_header.erb` a lot too.

You can also drop the `if $headers` and stick to `! empty($headers)` on line 2263.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: manifests/vhost.pp (hunk 1 above)
Comment (chillinger) [reply manifests/vhost.pp:1856]: @ekohl  I thought about that, but realised the patch the way I did for the following reasons:
- I only have the time to actually fix the bug which breaks our puppet runs, I did not want to rework more than really necessary
- every other array type parameter in the module, including $request_headers is an Optional[Array[String]]. I wanted to stay consistent

For those reasons I would like this fix to be accepted. If the datatype should be changed to Array with default [], I would like this seen as an independent rework issue

**Comment by david22swan** [comment]
@chillinger Apologies, but after some discussion within the team we have decided to go ahead with the changes proposed by @ekohl rather than your own, as they are more comprehensive and represent a larger improvement to the module.
From what I can see they should still solve your issue and we hope to get them merged in soon.
I am sorry for any inconvenience this has caused you, and while we could not merge your work at this time we hope to see more from you in the future ad thank you for bringing this issue to our attention.

**Comment by chillinger** [comment]
@david22swan @ekohl  no problem at all, I'm happy an even better solution arose from my initial proposal!

**Pull request closed without merging**